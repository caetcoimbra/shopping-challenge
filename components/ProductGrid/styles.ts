import styled from "styled-components";

export const Container = styled.ul`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: 18px auto 50px auto;
  max-width: 1200px;
  gap: 18px;
`;
