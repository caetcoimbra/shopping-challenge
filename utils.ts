export const formatBrlCurrency = (value: number) =>
value
  .toLocaleString("pt-br", { style: "currency", currency: "BRL" })
  .slice(0, -3);