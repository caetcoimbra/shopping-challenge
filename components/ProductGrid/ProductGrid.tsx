import ProductCard from "@/components/ProductCard/ProductCard";
import { Container } from "./styles";
import { Product } from "../../integrations/request";

export interface ProductsProps {
  products: Product[];
  loading: boolean;
  skeletonsCount?: number;
}

const ProductGrid = ({
  products,
  loading,
  skeletonsCount = 8,
}: ProductsProps) => {
  return (
    <Container>
      {loading
        ? Array.from({ length: skeletonsCount }).map((_, i) => (
            <ProductCard.Skeleton key={i} />
          ))
        : products.map((product) => (
            <li key={product.id}>
              <ProductCard {...product}></ProductCard>
            </li>
          ))}
    </Container>
  );
};

export default ProductGrid;
