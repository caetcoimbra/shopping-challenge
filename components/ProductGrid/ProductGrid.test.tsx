import { render } from "@testing-library/react";
import ProductGrid from "@/components/ProductGrid/ProductGrid";
import { mockProducts, mainTheme } from "../../tests/utils";

it("skeletons must be rendered", () => {
  const { container } = render(<ProductGrid products={[]} loading />);
  expect(container).toMatchSnapshot();
});

it("products must be rendered", () => {
  const { container } = render(
    mainTheme(<ProductGrid products={mockProducts} loading={false} />)
  );
  expect(container).toMatchSnapshot();
});
