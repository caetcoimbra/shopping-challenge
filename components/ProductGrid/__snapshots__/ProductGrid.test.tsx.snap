// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`products must be rendered 1`] = `
<div>
  <ul
    class="sc-kDvujY dGLUKA"
  >
    <li>
      <div
        class="sc-gswNZR innFoO"
      >
        <img
          alt="Iphone 11 128 GB"
          class="sc-dkrFOg kwQKsm"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/iphone11x128.webp"
        />
        <div
          class="sc-jSUZER hGltBr"
        >
          <h1
            class="sc-hLBbgP jsmdxh"
          >
            Iphone 11 128 GB
          </h1>
          <span
            class="sc-bcXHqe eWcSnT"
            style="height: 30px;"
          >
            R$ 5.000
          </span>
        </div>
        <p
          class="sc-eDvSVe jWkNbF"
        >
          Grave vídeos 4K, faça belos retratos e capture paisagens inteiras com o novo sistema de câmera dupla.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKPRtg vlEqf"
        >
          <svg
            class="sc-iBYQkv cREeEt"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
    <li>
      <div
        class="sc-gswNZR innFoO"
      >
        <img
          alt="AirPods"
          class="sc-dkrFOg kwQKsm"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/airpods.webp"
        />
        <div
          class="sc-jSUZER hGltBr"
        >
          <h1
            class="sc-hLBbgP jsmdxh"
          >
            AirPods
          </h1>
          <span
            class="sc-bcXHqe eWcSnT"
            style="height: 30px;"
          >
            R$ 1.200
          </span>
        </div>
        <p
          class="sc-eDvSVe jWkNbF"
        >
          Criados pela Apple Ligam e se conectam automaticamente Configuração com apenas um toque para todos seus aparelhos Apple.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKPRtg vlEqf"
        >
          <svg
            class="sc-iBYQkv cREeEt"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
    <li>
      <div
        class="sc-gswNZR innFoO"
      >
        <img
          alt="Macbook Air"
          class="sc-dkrFOg kwQKsm"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/macbookair.webp"
        />
        <div
          class="sc-jSUZER hGltBr"
        >
          <h1
            class="sc-hLBbgP jsmdxh"
          >
            Macbook Air
          </h1>
          <span
            class="sc-bcXHqe eWcSnT"
            style="height: 30px;"
          >
            R$ 8.200
          </span>
        </div>
        <p
          class="sc-eDvSVe jWkNbF"
        >
          Processador intel Core i5 de dois núcleos e 1,8 GHz (Turbo Boost de até 2,9 GHz) com cache L3 compartilhado de 3 MB.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKPRtg vlEqf"
        >
          <svg
            class="sc-iBYQkv cREeEt"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
    <li>
      <div
        class="sc-gswNZR innFoO"
      >
        <img
          alt="iPhone 12 64 GB"
          class="sc-dkrFOg kwQKsm"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/iphone12x64.webp"
        />
        <div
          class="sc-jSUZER hGltBr"
        >
          <h1
            class="sc-hLBbgP jsmdxh"
          >
            iPhone 12 64 GB
          </h1>
          <span
            class="sc-bcXHqe eWcSnT"
            style="height: 30px;"
          >
            R$ 6.500
          </span>
        </div>
        <p
          class="sc-eDvSVe jWkNbF"
        >
          Grave vídeos 4K, faça belos retratos e capture paisagens inteiras com o novo sistema de câmera dupla.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKPRtg vlEqf"
        >
          <svg
            class="sc-iBYQkv cREeEt"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
    <li>
      <div
        class="sc-gswNZR innFoO"
      >
        <img
          alt="Apple Watch Series 7"
          class="sc-dkrFOg kwQKsm"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/applewatch-series7.webp"
        />
        <div
          class="sc-jSUZER hGltBr"
        >
          <h1
            class="sc-hLBbgP jsmdxh"
          >
            Apple Watch Series 7
          </h1>
          <span
            class="sc-bcXHqe eWcSnT"
            style="height: 30px;"
          >
            R$ 3.200
          </span>
        </div>
        <p
          class="sc-eDvSVe jWkNbF"
        >
          O Apple Watch faz coisas que outros aparelhos não conseguem porque ele fica no seu pulso.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKPRtg vlEqf"
        >
          <svg
            class="sc-iBYQkv cREeEt"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
    <li>
      <div
        class="sc-gswNZR innFoO"
      >
        <img
          alt="iPad"
          class="sc-dkrFOg kwQKsm"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/ipadair.webp"
        />
        <div
          class="sc-jSUZER hGltBr"
        >
          <h1
            class="sc-hLBbgP jsmdxh"
          >
            iPad
          </h1>
          <span
            class="sc-bcXHqe eWcSnT"
            style="height: 30px;"
          >
            R$ 4.200
          </span>
        </div>
        <p
          class="sc-eDvSVe jWkNbF"
        >
          iPad é uma linha de tablets projetada, desenvolvida e comercializada pela Apple, que funciona com o sistema operacional móvel iOS e iPadOS.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKPRtg vlEqf"
        >
          <svg
            class="sc-iBYQkv cREeEt"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
    <li>
      <div
        class="sc-gswNZR innFoO"
      >
        <img
          alt="Headset Cloud Revolver"
          class="sc-dkrFOg kwQKsm"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/hyperx-cloudrevolver.webp"
        />
        <div
          class="sc-jSUZER hGltBr"
        >
          <h1
            class="sc-hLBbgP jsmdxh"
          >
            Headset Cloud Revolver
          </h1>
          <span
            class="sc-bcXHqe eWcSnT"
            style="height: 30px;"
          >
            R$ 1.000
          </span>
        </div>
        <p
          class="sc-eDvSVe jWkNbF"
        >
          A linha HyperX Cloud Revolver foi projetada para atender as exigências dos gamers de PC ou de console.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKPRtg vlEqf"
        >
          <svg
            class="sc-iBYQkv cREeEt"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
    <li>
      <div
        class="sc-gswNZR innFoO"
      >
        <img
          alt="Headset Cloud Stinger"
          class="sc-dkrFOg kwQKsm"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/hyperxcloudstinger.webp"
        />
        <div
          class="sc-jSUZER hGltBr"
        >
          <h1
            class="sc-hLBbgP jsmdxh"
          >
            Headset Cloud Stinger
          </h1>
          <span
            class="sc-bcXHqe eWcSnT"
            style="height: 30px;"
          >
            R$ 600
          </span>
        </div>
        <p
          class="sc-eDvSVe jWkNbF"
        >
          O HyperX Cloud Stinger™ é o headset ideal para jogadores que procuram leveza e conforto, qualidade de som superior e maior praticidade.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKPRtg vlEqf"
        >
          <svg
            class="sc-iBYQkv cREeEt"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
  </ul>
</div>
`;

exports[`skeletons must be rendered 1`] = `
<div>
  <ul
    class="sc-kDvujY dGLUKA"
  >
    <div
      class="sc-gswNZR sc-jrcTuL innFoO biTDni"
    >
      <div
        class="sc-ftTHYK hKcRTi"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
    </div>
    <div
      class="sc-gswNZR sc-jrcTuL innFoO biTDni"
    >
      <div
        class="sc-ftTHYK hKcRTi"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
    </div>
    <div
      class="sc-gswNZR sc-jrcTuL innFoO biTDni"
    >
      <div
        class="sc-ftTHYK hKcRTi"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
    </div>
    <div
      class="sc-gswNZR sc-jrcTuL innFoO biTDni"
    >
      <div
        class="sc-ftTHYK hKcRTi"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
    </div>
    <div
      class="sc-gswNZR sc-jrcTuL innFoO biTDni"
    >
      <div
        class="sc-ftTHYK hKcRTi"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
    </div>
    <div
      class="sc-gswNZR sc-jrcTuL innFoO biTDni"
    >
      <div
        class="sc-ftTHYK hKcRTi"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
    </div>
    <div
      class="sc-gswNZR sc-jrcTuL innFoO biTDni"
    >
      <div
        class="sc-ftTHYK hKcRTi"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
    </div>
    <div
      class="sc-gswNZR sc-jrcTuL innFoO biTDni"
    >
      <div
        class="sc-ftTHYK hKcRTi"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
      <div
        class="sc-pyfCe fKgnaB"
      />
    </div>
  </ul>
</div>
`;
